tests/providers/google/cloud/hooks/test_dataform.py

```diff
@@ -0,0 +1,220 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from unittest import TestCase, mock
+
+from google.api_core.gapic_v1.method import DEFAULT
+
+from airflow.providers.google.cloud.hooks.dataform import DataformHook
+from tests.providers.google.cloud.utils.base_gcp_mock import mock_base_gcp_hook_default_project_id
+
+BASE_STRING = "airflow.providers.google.common.hooks.base_google.{}"
+DATAFORM_STRING = "airflow.providers.google.cloud.hooks.dataform.{}"
+
+PROJECT_ID = "project-id"
+REGION = "region"
+REPOSITORY_ID = "test_repository"
+WORKSPACE_ID = "test_workspace"
+GCP_CONN_ID = "google_cloud_default"
+DELEGATE_TO = "test-delegate-to"
+IMPERSONATION_CHAIN = ["ACCOUNT_1", "ACCOUNT_2", "ACCOUNT_3"]
+COMPILATION_RESULT = {
+    "git_commitish": "main",
+    "workspace": (
+        f"projects/{PROJECT_ID}/locations/{REGION}/repositories/{REPOSITORY_ID}/workspaces/{WORKSPACE_ID}"
+    ),
+}
+COMPILATION_RESULT_ID = "test_compilation_result_id"
+WORKFLOW_INVOCATION = {
+    "compilation_result": (
+        f"projects/{PROJECT_ID}/locations/{REGION}/repositories/"
+        f"{REPOSITORY_ID}/compilationResults/{COMPILATION_RESULT_ID}"
+    ),
+}
+WORKFLOW_INVOCATION_ID = "test_workflow_invocation_id"
+
+
+class TestDataflowHook(TestCase):
```

Comment:
Should we use unittest? CONTRIBUTING and TESTING docs mention that we should use pytest onwards, and rewrite existing classes from unittest.

Reply:
We need to upgrade generator for this files to use new approach